From 6eca1e7a830b5937ea0ba3ab91e2d89fdcdc3b22 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:46:20 -0400
Subject: [pci] Assign CardBus bus number only during the second pass

Message-id: <20140806034620.1438.60388.stgit@gir.stowe>
Patchwork-id: 87430
O-Subject: [RHEL7.1 PATCH 21/69] PCI: Assign CardBus bus number only during the second pass
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 619c8c310f7f21e59a7e2b53795183c34401a599

commit 619c8c310f7f21e59a7e2b53795183c34401a599
Author: Andreas Noever <andreas.noever@gmail.com>
Date:   Thu Jan 23 21:59:23 2014 +0100

    PCI: Assign CardBus bus number only during the second pass

    Right now the CardBus code in pci_scan_bridge() is executed during both
    passes. Since we always allocate the bus number ourselves it makes sense
    to put it into the second pass.

    Signed-off-by: Andreas Noever <andreas.noever@gmail.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index 9324a44..590afb1 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -874,7 +874,7 @@ int pci_scan_bridge(struct pci_bus *bus, struct pci_dev *dev, int max, int pass)
    * do in the second pass.
    */
   if (!pass) {
-   if (pcibios_assign_all_busses() || broken)
+   if (pcibios_assign_all_busses() || broken || is_cardbus)
     /* Temporarily disable forwarding of the
        configuration cycles on all bridges in
        this bus segment to avoid possible
-- 
1.7.1